### ISSUE TITLE
Further improvements to BuiltInKeywords test needed

### ISSUE THREAD

**Issue by michaelhkay** [qt4cg/qtspecs#1368]
1. The test generated for fn:seconds() is incorrect (can't handle required type xs:decimal?)
2. An incorrect argument is supplied for fn:pin()

**Comment by ChristianGruen** [comment]
I’m just adding this (sorry, I think you are aware of it anyway): `Keywords-fn-do-until-1` still causes an endless loop.

**Comment by michaelhkay** [comment]
I'm proposing to change the generated tests so they take the form:

```
<test-case name="Keywords-fn-char-1">
      <description>Test of keyword arguments in static function call</description>
      <created by="generate-keyword-test-set.xsl" on="2024-08-16+01:00"/>
      <environment ref="ka"/>
      <test>fn:char(value := ?) instance of function((xs:string | xs:positiveInteger)) as xs:string</test>
      <result>
         <assert-true/>
      </result>
   </test-case>
```
This (a) tests that the argument keywords from the spec are recognised, and (b) that the function signature is correctly implemented, without the complexity of trying to choose argument expressions that make sense for each function.

**Comment by michaelhkay** [comment]
Closed by PR #1390
